### PR TITLE
Integrate TRT-LLM v0.11

### PR DESCRIPTION
**Pull request by oyilmaz-nvidia**
# What does this PR do ?

Integrates TRT-LLM v0.11 to nemo export module.

### CODE REVIEW

File: nemo/export/tensorrt_llm.py [hunk 1]
```diff
@@ -38,16 +38,24 @@
...
+LOGGER = logging.getLogger("NeMo")
+
+use_model_opt = True
+try:
+    from nemo.export.trt_llm.qnemo import qnemo_to_tensorrt_llm
+    from nemo.export.trt_llm.qnemo.tokenizer_utils import get_nmt_tokenizer
+    from nemo.export.trt_llm.qnemo.utils import is_qnemo_checkpoint
+except Exception as e:
+    LOGGER.warning(f"Cannot import the Model Optimizer, it will not be available. {type(e).__name__}: {e}")
+    use_model_opt = False
```
Comment (janekl) [inline nemo/export/tensorrt_llm.py:53]: Is this import guard really necessary? modelopt should be available in the container.

File: nemo/export/tensorrt_llm.py [hunk 2]
```diff
@@ -38,16 +38,24 @@
...
 )
-from nemo.export.trt_llm.qnemo import qnemo_to_tensorrt_llm
-from nemo.export.trt_llm.qnemo.tokenizer_utils import get_nmt_tokenizer
-from nemo.export.trt_llm.qnemo.utils import is_qnemo_checkpoint
 from nemo.export.trt_llm.tensorrt_llm_build import build_and_save_engine
 from nemo.export.trt_llm.tensorrt_llm_run import generate, generate_streaming, load, load_distributed, refit
 
+LOGGER = logging.getLogger("NeMo")
+
+use_model_opt = True
```
Comment (janekl) [inline nemo/export/tensorrt_llm.py:46]: I'll prepare MR to remove modelopt dependency in `nemo.export`

File: tests/infer_data_path.py [hunk 3]
```diff
@@ -367,6 +264,17 @@ def get_infer_test_data():
     test_data["STARCODER1-15B-base"]["max_output_len"] = 128
     test_data["STARCODER1-15B-base"]["max_batch_size"] = 5
 
+    test_data["STARCODER2-15B-base"] = {}
+    test_data["STARCODER2-15B-base"]["model_type"] = "starcoder"
+    test_data["STARCODER2-15B-base"]["min_tps"] = 1
+    test_data["STARCODER2-15B-base"]["location"] = "Local"
+    test_data["STARCODER2-15B-base"]["model_dir"] = "/tmp/STARCODER2-15B-base/trt_llm_model-1/"
+    test_data["STARCODER2-15B-base"]["checkpoint"] = "/opt/checkpoints/starcoder-2_15b_4k_vfinal/4194b.nemo"
```
Comment (janekl) [inline tests/infer_data_path.py:272]: Naming convention is abused here. Could we make it consistent with other models from `/opt/checkpoints`?

Comment (janekl) [reply tests/infer_data_path.py:272]: There are these two models available, maybe we can use them here:
```
STARCODER2-15B-base
STARCODER2-15B-lc
```

File: nemo/export/tensorrt_llm.py (hunk 1 above)
Comment (oyilmaz-nvidia) [reply nemo/export/tensorrt_llm.py:53]: If we can make modelopt work, we don't need it. Needed it to make it work for this PR. Once the modelopt part is ready and working, that guard can be removed.

File: nemo/export/tensorrt_llm.py (hunk 2 above)
Comment (oyilmaz-nvidia) [reply nemo/export/tensorrt_llm.py:46]: Thanks Jan! You can remove that guard in your PR.

File: nemo/export/tensorrt_llm.py (hunk 1 above)
Comment (ericharper) [reply nemo/export/tensorrt_llm.py:53]: The import guard is needed unless modelopt can be made a hard requirement

**Review by meatybobby — APPROVED** [review]
LGTM

**Review by meatybobby — APPROVED** [review]

**Pull request merged**